File: src/types/project.ts
export interface BaselinePoint {
  id: string;
  name: string;
  type: "point" | "line" | "region";
  coordinates: number[];
  frameIndex: number;
}

export interface Project {
  id: string;
  name: string;
  createdAt: number;
  updatedAt: number;
  canvasWidth: number;
  canvasHeight: number;
  /**
   * 帧率仅作为播放速度标记和导入时的智能建议。
   * 不限制实际帧数——帧数完全由用户管理。
   */
  fps: number;
  baselinePoints: BaselinePoint[];
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct BaselinePoint {
    pub id: String,
    pub name: String,
    #[serde(rename = "type")]
    pub point_type: String,
    pub coordinates: Vec<f64>,
    pub frame_index: i64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct Project {
    pub id: String,
    pub name: String,
    pub canvas_width: i64,
    pub canvas_height: i64,
    pub fps: i64,
    pub created_at: i64,
    pub updated_at: i64,
    #[serde(default)]
    pub baseline_points: Vec<BaselinePoint>,
}
